autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -132,6 +140,11 @@ protected void execute() {
             }
         } catch (Exception ex) {
             LOGGER.info("Failed to process load alert for Cluster '{}', userCrn: {}, exception: ", stackCrn, pollingUserCrn, ex);
+            if (ex instanceof ForbiddenException) {
+                metricService.incrementMetricCounter(YARN_FORBIDDEN_EXCEPTION, MetricTag.TENANT.name(),
+                        Optional.ofNullable(cluster.getClusterPertain().getTenant()).orElse("NA"), MetricTag.CLOUD_PROVIDER.name(),
+                        Optional.ofNullable(cluster.getCloudPlatform()).orElse("NA"));
+            }
```

Comment:
shouldn't we have this in YarnMetricsClient since there is actual call happening. Here we are assuming that ForbiddenException will be propagated which can change in future. Also, metric should be generated at the same place where actual operation is being executed.

Reply:
There are CM and Yarn API calls being made in the corresponding try block, along with a CB API call to retrieve stackv4response. We want to process all possible exceptions thrown from those invocations to this catch block and publish an UpdateFailedEvent. Modifying this logic to detect the exception in YarnMetricsClient would require another exception to be re-thrown which will eventually be caught here. Additionally, the handler for UpdateFailedEvent (UpdateFailedHandler) has some logic that processes these Forbidden Exceptions.

Reply:
This should be in the YarnMetricsClient. Also separately for the CM invocations. The bit about the UpdateFailedHandler tracking a single error is something we should address in a separate jira. (The invocation of YARN vs CM uses a different user; it doesn't make sense to change this user if the CM API throws a 403).

Additionally, we want to track 1) Overall errors from these API invocations, and 2) sub-categorize the 403 errors since those are the most problematic right now. (Not sure if we can extract the error code and use that as a tag value, or we need to have a separate metric/tag based on the ForbiddenException)

Reply:
Moved this to YarnMetricsClient with exception handling